testsuite/model/src/test/java/org/keycloak/testsuite/model/KeycloakModelTest.java

```diff
@@ -302,8 +302,11 @@ protected void finished(Description description) {
      * local to the thread that calls this method, allowing for per-thread customization. This in turn allows
      * testing of several parallel session factories which can be used to simulate several servers
      * running in parallel.
+     * <p>
+     * The method itself is synchronized as <code>CONFIG</code> and <code>MODEL_PARAMETERS</code> are shared resources
+     * used during the initialization of the factories.
      */
-    public static KeycloakSessionFactory createKeycloakSessionFactory() {
+    public static synchronized KeycloakSessionFactory createKeycloakSessionFactory() {
```

Comment:
Does this effectively mean that `inIndependentFactories()` will now spawn new nodes sequentially? And if so, do we want it?

Reply:
Yes, @martin-kanis that was my intention when I created this PR as I thought CONFIG wouldn't be thread safe. I now looked at it again, and saw that will use a thread safe implementation when it is not running on the main thread. Therefore I reverted that part, and only keep the retry login the updated PR. Please let me know if this now looks better to you and can be approved.

Reply:
Thanks for the explanation and changes. I scheduled [a run in GHA](https://github.com/martin-kanis/keycloak/actions/runs/4415190525/jobs/7737848284) with 10 model test executions to see if we get any retry triggered.